"lxc image list" crashes if first column has duplicates

```
stgraber@castiana:~/data/code/lxc/lxd (stgraber/new-client)$ lxc image list -clf
panic: runtime error: index out of range

goroutine 1 [running]:
main.(*SortImage).Less(0xc42015cf40, 0x2, 0x1, 0x1)
	<autogenerated>:3 +0x127
sort.insertionSort(0xc78a60, 0xc42015cf40, 0x0, 0x5)
	/usr/lib/go-1.8/src/sort/sort.go:29 +0x71
sort.quickSort(0xc78a60, 0xc42015cf40, 0x0, 0x5, 0x6)
	/usr/lib/go-1.8/src/sort/sort.go:211 +0x1eb
sort.Sort(0xc78a60, 0xc42015cf40)
	/usr/lib/go-1.8/src/sort/sort.go:220 +0x79
main.(*imageCmd).showImages.func1(0xc42016ac60, 0xc42015ce40, 0x2)
	/home/stgraber/data/code/go/src/github.com/lxc/lxd/lxc/image.go:985 +0x4ee
main.(*imageCmd).showImages(0xc420077bc0, 0xc420174700, 0x5, 0x8, 0xc4200c2f80, 0x0, 0x0, 0xc420114ff0, 0x2, 0x2, ...)
	/home/stgraber/data/code/go/src/github.com/lxc/lxd/lxc/image.go:1007 +0x8c1
main.(*imageCmd).run(0xc420077bc0, 0xc420077dc0, 0xc42011f830, 0x1, 0x1, 0x0, 0x0)
	/home/stgraber/data/code/go/src/github.com/lxc/lxd/lxc/image.go:779 +0x5db2
main.run(0xc420098a00, 0xc420075e30)
	/home/stgraber/data/code/go/src/github.com/lxc/lxd/lxc/main.go:157 +0x74d
main.main()
	/home/stgraber/data/code/go/src/github.com/lxc/lxd/lxc/main.go:27 +0x6e
```

Looks like SortImage needs to be reworked to work with the new "-c" argument of "lxc image list".

--- a/lxc/config.go
+++ b/lxc/config.go
@@ -310,7 +310,7 @@ func (c *configCmd) run(config *lxd.Config, args []string) error {
 				i18n.G("COMMON NAME"),
 				i18n.G("ISSUE DATE"),
 				i18n.G("EXPIRY DATE")})
-			sort.Sort(SortImage(data))
+			sort.Sort(StringList(data))
 			table.AppendBulk(data)
 			table.Render()
 

--- a/lxc/image.go
+++ b/lxc/image.go
@@ -744,7 +744,7 @@ func (c *imageCmd) showImages(images []api.Image, filters []string, columns []im
 			data = append(data, row)
 		}
 
-		sort.Sort(SortImage(data))
+		sort.Sort(StringList(data))
 		return data
 	}
 
@@ -814,7 +814,7 @@ func (c *imageCmd) showAliases(aliases []api.ImageAliasesEntry, filters []string
 		i18n.G("ALIAS"),
 		i18n.G("FINGERPRINT"),
 		i18n.G("DESCRIPTION")})
-	sort.Sort(SortImage(data))
+	sort.Sort(StringList(data))
 	table.AppendBulk(data)
 	table.Render()
 

--- a/lxc/utils.go
+++ b/lxc/utils.go
@@ -74,39 +74,33 @@ func (p *ProgressRenderer) UpdateOp(op api.Operation) {
 	}
 }
 
-// Image fingerprint and alias sorting
-type SortImage [][]string
+type StringList [][]string
 
-func (a SortImage) Len() int {
+func (a StringList) Len() int {
 	return len(a)
 }
 
-func (a SortImage) Swap(i, j int) {
+func (a StringList) Swap(i, j int) {
 	a[i], a[j] = a[j], a[i]
 }
 
-func (a SortImage) Less(i, j int) bool {
-	if a[i][0] == a[j][0] {
-		if a[i][3] == "" {
-			return false
+func (a StringList) Less(i, j int) bool {
+	x := 0
+	for x = range a[i] {
+		if a[i][x] != a[j][x] {
+			break
 		}
-
-		if a[j][3] == "" {
-			return true
-		}
-
-		return a[i][3] < a[j][3]
 	}
 
-	if a[i][0] == "" {
+	if a[i][x] == "" {
 		return false
 	}
 
-	if a[j][0] == "" {
+	if a[j][x] == "" {
 		return true
 	}
 
-	return a[i][0] < a[j][0]
+	return a[i][x] < a[j][x]
 }
 
 // Container name sorting

--- a/lxc/utils_test.go
+++ b/lxc/utils_test.go
@@ -1,0 +1,37 @@
+package main
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type utilsTestSuite struct {
+	suite.Suite
+}
+
+func TestUtilsTestSuite(t *testing.T) {
+	suite.Run(t, new(utilsTestSuite))
+}
+
+// StringList can be used to sort a list of strings.
+func (s *utilsTestSuite) Test_StringList() {
+	data := [][]string{{"foo", "bar"}, {"baz", "bza"}}
+	sort.Sort(StringList(data))
+	s.Equal([][]string{{"baz", "bza"}, {"foo", "bar"}}, data)
+}
+
+// The first different string is used in sorting.
+func (s *utilsTestSuite) Test_StringList_sort_by_column() {
+	data := [][]string{{"foo", "baz"}, {"foo", "bar"}}
+	sort.Sort(StringList(data))
+	s.Equal([][]string{{"foo", "bar"}, {"foo", "baz"}}, data)
+}
+
+// Empty strings are sorted last.
+func (s *utilsTestSuite) Test_StringList_empty_strings() {
+	data := [][]string{{"", "bar"}, {"foo", "baz"}}
+	sort.Sort(StringList(data))
+	s.Equal([][]string{{"foo", "baz"}, {"", "bar"}}, data)
+}

